src/main/java/xbot/common/subsystems/drive/PurePursuitCommand.java

```diff
@@ -337,10 +343,24 @@ private void advancePointIfNotAtLastPoint() {
      * @param translation The desired translation power, which will be constrained by the budget.
      * @return 
      */
-    private double constrainToMotionBudget(double rotation, double translation) {
-        double remainingBudget = motionBudget.get() - Math.abs(rotation);
-        double constrainedRemainingBudget = MathUtils.constrainDouble(remainingBudget, 0, 1);
-        return translation * constrainedRemainingBudget;
+    private double constrainTranslation(double rotation, double translation, double dotProduct) {
+        if (useDotProductDriving) {
+            return translation *= dotProduct;
+        } else {
+            double remainingBudget = motionBudget.get() - Math.abs(rotation);
```

Comment:
In theory now that we're not doing these steps, how is the blending of translation and rotation handled by the drive, is it doing our 'standard' arcade drive math? Eg if it wants to translate at -1.0 for facing the wrong direction but also turn 1.0?